### PR TITLE
Update the Windows build notes

### PR DESCRIPTION
**Pull request by droark**
- Clarify the Windows Subsystem For Linux section.
- Clarify the build requirements.

### CODE REVIEW

**Comment by fanquake** [comment]
Looks ok utACK 1f30bc3

File: doc/build-windows.md [hunk 1]
```diff
@@ -68,7 +72,7 @@ Then build using:
     cd ..
     ./autogen.sh # not required when building from tarball
     CONFIG_SITE=$PWD/depends/x86_64-w64-mingw32/share/config.site ./configure --prefix=/
-    make
+    make -j4
```
Comment (jonasschnelli) [inline doc/build-windows.md]: nit: not sure if we should set the `jobs` amount in the documentation. IMO we also don't do that for other build docs. It depends on available RAM/cores.

Comment (sipa) [inline doc/build-windows.md]: Ff you're going to add `-j` in places, at least explain that it impacts parallellism in a comment.

**Comment by droark** [comment]
Removed the -j flags (my own and the ones that were present) and did a fixup/reword. @jonasschnelli and @sipa brought up good points regarding parallelism. Doing a parallel make on the depends portion caused several errors on my system. I was able to keep trying and get it to work but it wasn't ideal.

**Comment by droark** [comment]
Also, just to clarify, I did successfully build and run the 32-bit version. `make check` passed too.

**Comment by MarcoFalke** [comment]
ACK b705a04

**Pull request merged**